src/sodaspark/scan.py

```diff
@@ -297,3 +301,136 @@ def execute(
     scan = create_scan(scan_yml, soda_server_client=soda_server_client)
     scan.execute()
     return scan.scan_result
+
+
+def measurements_to_data_frame(measurements: list[Measurement]) -> DataFrame:
+    """
+    Convert measurements to a data frame.
+    Parameters
+    ----------
+    measurements: List[Measurement]
+        The measurements.
+    Returns
+    -------
+    out : DataFrame
+        The measurements as data frame.
+    """
+    schema_group_values = T.StructType(
+        [
+            T.StructField(
+                "group", T.MapType(T.StringType(), T.StringType()), True
+            ),
+            T.StructField("value", T.StringType(), True),
+        ]
+    )
+    schema = T.StructType(
+        [
+            T.StructField("metric", T.StringType(), True),
+            T.StructField("columnName", T.StringType(), True),
+            T.StructField("value", T.StringType(), True),
+            T.StructField(
+                "groupValues", T.ArrayType(schema_group_values, True), True
+            ),
+        ]
+    )
+    spark_session = SparkSession.builder.getOrCreate()
+    out = spark_session.createDataFrame(
+        [measurement.to_dict() for measurement in measurements],
```

Comment:
could you remove the `to_dict` here? Just pas in measurements

Reply:
I have resolved this by updating the schema accordingly.